opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -1107,16 +1120,65 @@ groupControlledWells(const Schedule& schedule,
 
         if (included) {
             num_wells
-                += groupControlledWells(schedule, well_state, group_state, report_step, child_group, always_included_child, is_production_group, injection_phase);
+                += groupControlledWells(schedule, well_state, group_state, summary_state, report_step, child_group, always_included_child, is_production_group, injection_phase);
         }
     }
     for (const std::string& child_well : group.wells()) {
         bool included = (child_well == always_included_child);
         if (is_production_group) {
-            included = included || well_state.isProductionGrup(child_well);
+                included = included || well_state.isProductionGrup(child_well) || group.as_choke();
         } else {
             included = included || well_state.isInjectionGrup(child_well);
         }
+        const auto ctrl1 = group_state.production_control(group.name());
+        if (group.as_choke() && ((ctrl1 == Group::ProductionCMode::FLD) || (ctrl1 == Group::ProductionCMode::NONE))){
+            // The auto choke group has not own group control but inherits control from an ancestor group.
+            // Number of wells should be calculated as zero when wells of auto choke group do not deliver target.
+            // This behaviour is then similar to no-autochoke group with wells not on GRUP control.
+            // The rates of these wells are summed up. The parent group target is reduced with this rate.
+            // This reduced target becomes the target of the other child group of this parent.
+            const PhaseUsage& pu = well_state.phaseUsage();
+            std::vector<Scalar> rates(pu.num_phases, 0.0);
+            for (int phase_pos = 0; phase_pos < pu.num_phases; ++phase_pos) {
+                 rates[phase_pos] = WellGroupHelpers<Scalar>::sumWellSurfaceRates(group,
+                                                                                  schedule,
+                                                                                  well_state,
+                                                                                  report_step,
+                                                                                  phase_pos,
+                                                                                  false);
+            }
+
+            // Get the ancestor of the auto choke group that has group control (cmode != FLD, NONE)
+            const auto& control_group_name = control_group(group, group_state, report_step, schedule);
+            const auto& control_group = schedule.getGroup(control_group_name, report_step);
+            const auto& ctrl = control_group.productionControls(summary_state);
+            const auto& control_group_guide_rate = ctrl.guide_rate;
+            const auto& control_group_cmode = ctrl.cmode;
```

Comment:
This is fixed. If the guide rate is defaulted it will be calculated from the children guide rates. Thus for 
```
GCONPROD--  1    2     3     4       5     6     7       8       9       10 
'M5S' ORAT  8000  1*      1*    1*  'RATE'  'YES'   1*    ''    1*   1*   1*   3* /
'B1'  FLD  1*     1*      1*    1*  'RATE'  'YES'   35.0   'OIL'    1*   1*   1*   3* /
'C1'  FLD  1*    1*       1*    1*  'RATE'  'YES’   5.0   'OIL'    1*   1*   1*   3* / 
```
The guide rate of `M5S` will be internally calculated as 40.